GVFS/GVFS.UnitTests/Common/GVFSEnlistmentHealthTests.cs

```diff
@@ -0,0 +1,234 @@
+﻿using GVFS.Common;
+using GVFS.Tests.Should;
+using NUnit.Framework;
+using System.Collections.Generic;
+
+namespace GVFS.UnitTests.Common
+{
+    [TestFixture]
+    public class GVFSEnlistmentHealthTests
+    {
+        private readonly char sep = GVFSPlatform.GVFSPlatformConstants.PathSeparator;
+
+        private List<string> fauxGitFiles = new List<string>();
+        private List<string> fauxGitFolders = new List<string>();
+        private List<string> fauxPlaceholderFiles = new List<string>();
+        private List<string> fauxPlaceholderFolders = new List<string>();
+        private List<string> fauxModifiedPathsFiles = new List<string>();
+        private List<string> fauxModifiedPathsFolders = new List<string>();
+        private List<string> fauxSkipWorktreeFiles = new List<string>();
+        private GVFSEnlistmentHealthCalculator enlistmentHealthCalculator;
+        private GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData;
+
+        [TestCase]
+        public void OnlyOneHydratedDirectory()
+        {
+            this.fauxGitFiles.AddRange(new string[] { "A/1.js", "A/2.js", "A/3.js", "A/4.js", "B/1.js", "B/2.js", "B/3.js", "B/4.js", "C/1.js", "C/2.js", "C/3.js", "C/4.js" });
+            this.fauxPlaceholderFiles.AddRange(new string[] { "A/1.js", "A/2.js", "A/3.js" });
+
+            this.enlistmentHealthData = this.GenerateStatistics(string.Empty);
+
+            this.enlistmentHealthData.DirectoryHydrationLevels[0].Key.ShouldEqual("A");
+            this.enlistmentHealthData.DirectoryHydrationLevels[0].Value.ShouldEqual(0.75m);
+            this.enlistmentHealthData.DirectoryHydrationLevels[1].Value.ShouldEqual(0);

```

Comment:
Why do we not check the `Key` for indexes `1` and `2`?

Reply:
Tracking this in a larger GVFSEnlistmentHealhTests refactoring issue: https://github.com/microsoft/VFSForGit/issues/1299